src/price-spread-ratio/price-spread-ratio.spec.js

```diff
@@ -0,0 +1,261 @@
+const axios = require('axios');
+const BigNumber = require('bignumber.js');
+
+// mock response from FTX API
+const mockFtxResponse = {
+  status: 200,
+  data: {
+    result: {
+      price: 0,
+    },
+  },
+};
+
+// mock the axios module for FTX API calls
+jest.mock('axios');
+axios.get.mockResolvedValue(mockFtxResponse);
+
+// mock the ethers module
+jest.mock('ethers', () => ({
+  ...jest.requireActual('ethers'),
+  Contract: jest.fn(),
+}));
+const ethers = require('ethers');
+
+const WEI_PER_ETHER = new BigNumber(ethers.constants.WeiPerEther.toString());
+
+const {
+  createAlert,
+  provideHandleBlock,
+  provideInitialize,
+} = require('./price-spread-ratio');
+
+/* mock axios test */
+describe('mock axios GET request', () => {
+  it('should call axios.get and return a response', async () => {
+    mockFtxResponse.data.result.price = 42;
+    const response = await axios.get('https://...');
+    expect(axios.get).toHaveBeenCalledTimes(1);
+    expect(response.data.result.price).toEqual(42);
+  });
+});
+
+/* agent tests */
+describe('price spread ratio monitoring', () => {
+  let initialize;
+  let handleBlock;
+  let data;
+
+  // for the tests, we will keep the mocked FTX API price constant and change the mocked price
+  // returned by the perpetual finance contract
+  const mockFtxPrice = 100;
+  const upperLimitPercent = 1;
+  const lowerLimitPercent = -1;
+  const mockContractName = 'FAKE1';
+  const mockContractAddress = '0xFAKEADDRESS1';
+
+  beforeEach(async () => {
+    axios.get.mockClear();
+
+    // reset the price returned by the mocked FTX API
+    mockFtxResponse.data.result.price = mockFtxPrice;
+
+    // initialize the data structure
+    data = {
+      provider: {},
+      contractNames: [
+        mockContractName,
+      ],
+      contractsData: [
+        {
+          name: mockContractName,
+          address: mockContractAddress,
+          timeThresholdSeconds: BigInt(100),
+        },
+      ],
+    };
+    initialize = provideInitialize(data);
+    await initialize();
+
```

Comment:
so that's why you have those `if` statements in the initialize function. I'm more of a fan of first initializing then adding things later but this is probably more rightish maybe... after initialize is called you can set things on teh data object and it will affect the handler